client/components/Slide/Components/AudioResponse/Display.jsx

```diff
@@ -1,49 +1,119 @@
 import { type } from './type';
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { Button, Label, Icon } from 'semantic-ui-react';
+import MicRecorder from 'mic-recorder-to-mp3';
+import { detect } from 'detect-browser';
 
 import './AudioResponse.css';
 
-const Display = ({ prompt, isRecording = false, onClickResponse }) => {
-    return (
-        <React.Fragment>
-            <Label>
-                <p>{prompt}</p>
-                {!isRecording && (
-                    <Button
-                        basic
-                        color="black"
-                        toggle
-                        onClick={onClickResponse}
-                    >
-                        <Icon
-                            name="circle"
-                            aria-label="Record an Audio Response"
-                        />
-                        Record Response
-                    </Button>
-                )}
-                {isRecording && (
-                    <Button basic negative onClick={onClickResponse}>
-                        <Icon
-                            name="stop circle"
-                            aria-label="Record an Audio Response"
-                        />
-                        Stop Recording
-                    </Button>
-                )}
-            </Label>
-        </React.Fragment>
-    );
-};
+const SUPPORTED_BROWSERS = ['chrome', 'firefox'];
+
+class Display extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isRecording: false,
+            type: '',
+            blobURL: ''
+        };
+
+        this.mp3Recorder = new MicRecorder({ bitRate: 128 });
+
+        this.onClick = this.onClick.bind(this);
+        this.onStart = this.onStart.bind(this);
+        this.onStop = this.onStop.bind(this);
+
+        this.browserSupported = SUPPORTED_BROWSERS.includes(detect().name);
+    }
+
+    onClick() {
+        this.setState(prevState => ({ isRecording: !prevState.isRecording }));
+    }
+
+    async onStart() {
+        await this.mp3Recorder.start();
+        this.setState({ isRecording: true });
+    }
+
+    async onStop() {
+        const [buffer, blob] = await (await this.mp3Recorder.stop()).getMp3();
+        const blobURL = URL.createObjectURL(blob);
+
+        const file = new File(buffer, 'recording.mp3', {
+            type: blob.type,
+            lastModified: Date.now()
+        });
+
+        let fd = new FormData();
```

Comment:
I'd go with `body` so it can be used as: 

```js
        const { s3Location } = await (await fetch('/api/media/audio', {
            method: 'POST',
            body
        })).json();

```